src/legacy/core_plugins/table_vis/public/agg_table/agg_table.js

```diff
@@ -138,34 +140,90 @@ uiModules
+                }
+                case 'avg': {
                   if (!isDate) {
-                    formattedColumn.total = formatter.convert(sum(table.rows) / table.rows.length);
+                    const total = sum(table.rows) / table.rows.length;
+                    formattedColumn.formattedTotal = formatter.convert(total);
+                    formattedColumn.total = total;
                   }
                   break;
-                case 'min':
-                  formattedColumn.total = formatter.convert(_.chain(table.rows).map(col.id).min().value());
+                }
+                case 'min': {
+                  const total = _.chain(table.rows).map(col.id).min().value();
+                  formattedColumn.formattedTotal = formatter.convert(total);
+                  formattedColumn.total = total;
                   break;
-                case 'max':
-                  formattedColumn.total = formatter.convert(_.chain(table.rows).map(col.id).max().value());
+                }
+                case 'max': {
+                  const total = _.chain(table.rows).map(col.id).max().value();
+                  formattedColumn.formattedTotal = formatter.convert(total);
+                  formattedColumn.total = total;
                   break;
-                case 'count':
-                  formattedColumn.total = table.rows.length;
+                }
+                case 'count': {
+                  const total = table.rows.length;
+                  formattedColumn.formattedTotal = formatter.convert(total);
+                  formattedColumn.total = total;
                   break;
+                }
                 default:
                   break;
               }
             }
 
             return formattedColumn;
           }).filter(column => column);
+
+          if (showPercentage) {
+            const insertAtIndex = _.findIndex($scope.formattedColumns, { title: percentageCol });
+
+            // column to show percentage for was removed
+            if (insertAtIndex < 0) return;
+
+            const { cols, rows } = addPercentageCol($scope.formattedColumns, percentageCol, table.rows, insertAtIndex);
+            $scope.rows = rows;
+            $scope.formattedColumns = cols;
+          }
         });
       }
     };
   });
+
+/**
+ * @param {[]Object} columns - the formatted columns that will be displayed
+ * @param {String} title - the title of the column to add to
+ * @param {[]Object} rows - the row data for the columns
+ * @param {Number} insertAtIndex - the index to insert the percentage column at
+ * @returns {Object} - cols and rows for the table to render now included percentage column(s)
+ */
+function addPercentageCol(columns, title, rows, insertAtIndex) {
+  const { id, total } = columns[insertAtIndex];
+  const newId = `${id}-percents`;
+  const formatter = getFormat({ id: 'percent' });
+  const newCols = insert(columns, insertAtIndex, {
+    title: `${title} Percentages`,
```

Comment:
We should add i18n here: call the JS function `i18n.translate()` from the `@kbn/i18n` package.
```js
i18n.translate('tableVis.params.percentageTableColumnName',{
  defaultMessage: '{title} Percentages',
  values: {
    title: name,
  },
})
```